submissions/lalalilulu/tiny-world-pre-oop/index.js

```diff
@@ -0,0 +1,77 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/lalalilulu/a-tiny-JS-world
+   Web app: https://lalalilulu.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+
+    constructor(species, name, gender, saying, friends = []) {
+        this.species = species;
+        this.name = name;
+        this.gender = gender;
+        this.saying = saying;
+        this.friends = Array.from(friends).join(", ");
+    }
+
+    introduce() {
+        return`${this.saying}, I am a ${this.species}, my name is ${this.name}, my gender is ${this.gender}. My friends: ${this.friends}`;
+    }
+}
+
+class Human extends Inhabitant{
+
+    constructor(name, gender, saying, friends, legs = 2, hands = 2) {
+        super('human', name, gender, saying, friends);
+        this.legs = legs;
+        this.hands = hands;
+    }
+
+    introduce() {
+        return  super.introduce() + ` and I have ${this.legs} legs and ${this.hands} hands`;
+    }
+}
+
+class Animal extends Inhabitant{
+
+    constructor(species, name, gender, saying, friends, legs = 4) {
+        super(species, name, gender, saying, friends);
+        this.legs = legs;
+    }
+
+    introduce() {
```

Comment:
Not required to fix now, but you may consider overloading super base class `Object.toString`. This method is implictly called in a string context. Function `print` ensures string context for its arguments, thus `print(inhabitant)` is effectively equivalent to `print(inhabitant.toString())`